atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionManager.java

```diff
@@ -38,7 +38,11 @@
         while (true) {
             checkOpen();
             try {
-                return runTaskThrowOnConflict(task);
+                T result = runTaskThrowOnConflict(task);
+                if (failureCount > 0) {
+                    log.info("Successfully completed transaction after {} retries.", failureCount);
```

Comment:
We should generate a unique transaction ID (say just a UUID but separate from its logical transaction timestamp) that we log as part of retry. 

Would also be useful to expose overall transaction manager metrics for retries (including rate of retry and time spent in retries), but that can happen later after #1356